### PR TITLE
feat(INT-177): intercom refactor

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by koladilip** [comment]
should we use cdkv2 ?

File: src/v0/destinations/intercom/util.js [hunk 1]
```diff
@@ -1,12 +1,215 @@
...
+ */
+const getBaseEndpoint = (destination) => {
+  const { apiServer } = destination.Config;
+  if (apiServer === 'eu') {
+    return BASE_EU_ENDPOINT;
+  }
+  if (apiServer === 'au') {
+    return BASE_AU_ENDPOINT;
+  }
+  return BASE_ENDPOINT;
```
Comment (utsabc) [inline src/v0/destinations/intercom/util.js:79]: use switch case

File: src/v0/destinations/intercom/util.js [hunk 2]
```diff
@@ -1,12 +1,215 @@
...
+
+  const processedUserResponse = processAxiosResponse(response);
+  if (isHttpStatusSuccess(processedUserResponse.status)) {
+    return processedUserResponse.response?.data.length > 0
+      ? processedUserResponse.response?.data[0]?.id
+      : null;
+  }
+
+  return null;
+};
```
Comment (utsabc) [inline src/v0/destinations/intercom/util.js:172]: We need to do 1 search api call for the group of events we are getting in router transform to make it efficient and pass the state to transformation

File: src/v0/destinations/intercom/transform.js [hunk 3]
```diff
@@ -1,251 +1,144 @@
...
-        customAttributes[key] = traits[key];
-      }
-    });
-  }
+const groupResponseBuilder = async (message, destination) => {
+  const payload = getPayload(message, ConfigCategory.GROUP);
+  payload.custom_attributes = filterCustomAttributes(payload, ReservedCompanyAttributes);
+  const companyId = await createOrUpdateCompany(payload, destination);
+  const lookupField = getLookUpField(message);
+  const contactId = await fetchContactId(message, destination, lookupField);
```
Comment (utsabc) [inline src/v0/destinations/intercom/transform.js:99]: Do this for all the inputs in router transform together for efficiency

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**